### DISCUSSION TITLE
[QUESTION] Multiple MessageHandlers with the same Filters

### DISCUSSION THREAD

**Discussion by sh-how** [python-telegram-bot/python-telegram-bot#2971] [Q&A] [answered]
### Issue I am facing

I have two messagehandlers that have the same regex filter but both are pointing to different functions. How do I disable one after it is called so that the other can be registered. Once I run part of the code that brings me to the first messagehandler e.g. (the one that directs to order), subsequent messagehandlers with the regex will all go to order and never to modify_confirm. I have tried using remove_handler but to no success.

### Related part of your code

```python
bot.handler = MessageHandler(Filters.regex(r'\d+\.*\d*\s\d+\.*\d*\s\d+\.*\d*'), order)
bot.dispatcher.add_handler(bot.handler)

bot.handler = MessageHandler(Filters.regex(r'\d+\.*\d*\s\d+\.*\d*\s\d+\.*\d*'), modify_confirm)
bot.dispatcher.add_handler(bot.handler)
```

### Operating System

Windows 10

### Version of Python, python-telegram-bot & dependencies

```shell
python-telegram-bot 13.11
Bot API 5.7
certifi 2021.10.08
```

**Comment by clot27** [comment]
Can you show what you have tried? and why you named both handlers as same "bot.handler" and why you specify handler as an attribute of bot.?

**Reply by sh-how** [reply]
The reason why I named both as bot.handler is because bot is an instance of a class. I thought that if I stored the handler in the class attribute, I would be able to access it outside of functions without passing it around.

**Comment by Bibo-Joshi** [comment]
For reference: This question has been asked in similar form at https://stackoverflow.com/questions/71986152

**Comment by sh-how** [comment, marked as answer]
I fixed it. What I did was remove my second MessageHandler declaration. And instead I updated the handler's callback attribute to the other function e.g. `bot.handler.callback = second_function`
